From 3f1704cd92c5ce9b3dc80545f428b975fbf97f91 Mon Sep 17 00:00:00 2001
From: Vasyl Gomonovych <vgomonovych@marvell.com>
Date: Sun, 9 Jan 2022 11:07:40 -0800
Subject: [PATCH 08/12] drivers: marvell: otx2-sdei-ghes: Mark driver ready
 handle SDEI

commit cf9955ab1c200c68599dee68d4e6d0b724ef940e from
git@git.assembla.com:cavium/WindRiver.linux.git

Setup shared buffer reg and allow firmware
deliver RAS SDEI records to kernel after
driver ready handle events.

Signed-off-by: Vasyl Gomonovych <vgomonovych@marvell.com>
Change-Id: I1e1cef84f76591a7da03f279133bc19a1ca611ce
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/68833
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Tested-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c | 4 +++-
 drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.h | 3 +++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c b/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c
index 9bb7ef96be8a..5fbef5ef3a5c 100644
--- a/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c
+++ b/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c
@@ -184,6 +184,7 @@ static int sdei_ghes_init(struct platform_device *pdev)
 				ret, event->id, event->name);
 			break;
 		}
+		event->ring->reg = OTX2_GHES_ERR_RING_SIG;
 	}
 
 	if (idx != ghes_drv->source_count) {
@@ -214,7 +215,7 @@ static int sdei_ghes_de_init(struct platform_device *pdev)
 
 	for (idx = 0; idx < ghes_drv->source_count; idx++) {
 		event = &ghes_drv->source_list[idx];
-
+		event->ring->reg = 0;
 		ret = sdei_event_disable(event->id);
 		if (ret < 0)
 			dev_err(dev,
@@ -734,6 +735,7 @@ static int __init sdei_ghes_bert_init(struct device_node *of_node)
 		ret = 0;
 		goto exit;
 	}
+	bed_ring->reg = OTX2_GHES_ERR_RING_SIG;
 
 	/*
 	 * The memory block contains the boot error data ring; beyond the ring
diff --git a/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.h b/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.h
index a89e95fec37e..9667b08fcf70 100644
--- a/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.h
+++ b/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.h
@@ -14,6 +14,7 @@
 #define __OTX2_SDEI_GHES_H__
 
 #define SDEI_GHES_EVENT_NAME_MAX_CHARS 16
+#define OTX2_GHES_ERR_RING_SIG    ((int)'M' << 24 | 'R' << 16 | 'V' << 8 | 'L')
 /*
  * Describes an error source per ACPI 18.3.2.6 (Generic Hardware Error Source).
  * This produces GHES-compliant error records from data forwarded by the [ATF]
@@ -69,6 +70,8 @@ struct otx2_ghes_err_ring {
 	uint32_t volatile head;
 	uint32_t          tail;
 	uint32_t          size;       /* ring size */
+	uint32_t sig;
+	uint32_t reg;
 	/* ring of records */
 	struct otx2_ghes_err_record records[1] __aligned(8);
 };
-- 
2.34.1

